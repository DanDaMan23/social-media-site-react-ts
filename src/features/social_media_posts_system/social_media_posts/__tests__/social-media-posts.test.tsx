import { fireEvent, render } from "@testing-library/react"
import {
  ISocialMediaPostsContext,
  SocialMediaPostsContext
} from "../../contexts/social-media-posts.context"
import SocialMediaPosts from "../social-media-posts"
import PostModel from "../../models/post.model"

const renderSocialMediaPosts = () => {
  const mockPosts = [
    new PostModel(1, "Hello World", "mockUser", new Date("2022-01-01"), []),
    new PostModel(2, "Hello World", "mockUser2", new Date("2022-01-02"), []),
    new PostModel(3, "Hello World", "mockUser3", new Date("2022-01-03"), [])
  ]

  const contextValue: ISocialMediaPostsContext = {
    posts: mockPosts,
    error: "",
    getNextPosts: jest.fn(),
    createNewPost: jest.fn(),
    hasMorePosts: false,
    createPostSuccess: "",
    createPostError: "",
    initialGetPosts: jest.fn()
  }

  return render(
    <SocialMediaPostsContext.Provider value={contextValue}>
      <SocialMediaPosts />
    </SocialMediaPostsContext.Provider>
  )
}

describe("social-media-posts test", () => {
  it("should render correctly", async () => {
    const { container } = renderSocialMediaPosts()

    expect(container).toMatchSnapshot()
  })
})
