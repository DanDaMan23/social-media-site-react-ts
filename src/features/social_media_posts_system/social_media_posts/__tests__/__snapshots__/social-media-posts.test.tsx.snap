// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`social-media-posts test should render correctly 1`] = `
<div>
  <div>
    <div
      class="vstack gap-3 minBreakpoint-xs"
    >
      <div
        class="card"
        style="background: rgb(208, 228, 234);"
      >
        <div
          class="card-body"
        >
          <div
            class="card-title h5"
          >
            mockUser
          </div>
          <p
            class="card-text"
          >
            Hello World
          </p>
          <small>
            December 31, 2021 6:00 PM
          </small>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <div
            class="card-title h5"
          >
            mockUser2
          </div>
          <p
            class="card-text"
          >
            Hello World
          </p>
          <small>
            January 1, 2022 6:00 PM
          </small>
        </div>
      </div>
      <div
        class="card"
        style="background: rgb(208, 228, 234);"
      >
        <div
          class="card-body"
        >
          <div
            class="card-title h5"
          >
            mockUser3
          </div>
          <p
            class="card-text"
          >
            Hello World
          </p>
          <small>
            January 2, 2022 6:00 PM
          </small>
        </div>
      </div>
    </div>
  </div>
</div>
`;
